labview/labview_rabbitmq.c

```diff
@@ -10,144 +10,268 @@
-        (*LVString)->cnt = len; // telling the Handle what string size to expect
-    }
-    return err;
+	int32 len = strlen(cpString);
+	MgErr err = NumericArrayResize(uB, 1, (UHandle*) &LVString, len);
+	if (!err)
+	{
+		strncpy((*LVString)->str, cpString, len);	// copying the string into the handle
+		(*LVString)->cnt = len;	// telling the Handle what string size to expect
+	}
+
+	return err;
+}
+
+MgErr copyBufferToLStrHandle(const void *buffer, int len, LStrHandle LVString)
+{
+	MgErr err = NumericArrayResize(uB, 1, (UHandle*) &LVString, len);
+	if (!err)
+	{
+		memcpy((*LVString)->str, buffer, len);	// copying the string into the handle
+		(*LVString)->cnt = len;	// telling the Handle what string size to expect
+	}
+
+																	
+			 
+	 
+																					
+																				
+	 
+	return err;
+}
+
+// Function to split a string into multiple strings based on the ";" separator
+char **splitString(char *input, int *count)
+{
+	char **tokens = NULL;
+	char *token = strtok(input, ";");
+	*count = 0;
+
+	while (token != NULL)
+	{
+		tokens = (char **) realloc(tokens, ((*count) + 1) *sizeof(char*));
+		if (tokens == NULL)
+		{
+			perror("Memory allocation error");
+			exit(1);
+		 
+										 
+				   
+								  
+		}
+
+		tokens[(*count)] = strdup(token);
+		(*count) ++;
+		token = strtok(NULL, ";");
+	}
+
+	return tokens;
+}
+
+void getConcatenatedMessageHeaders(amqp_table_t *table, LStrHandle cheaders_key, LStrHandle cheaders_value)
+{
+	// Calculate required buffer size for concatenated headers keys and values,
+	// following strings will be separated by ";". Buffer will be finished with Null-terminate the C string.
+	// Init counters with num of elements-1 (number of separators) + 1 ('\0')
+	int required_buffer_size_keys = (table->num_entries - 1) + 1;
+	int required_buffer_size_values = (table->num_entries - 1) + 1;
+	for (int i = 0; i < table->num_entries; i++)
+	{
+		required_buffer_size_keys += table->entries[i].key.len;
+		required_buffer_size_values += table->entries[i].value.value.bytes.len;
+	}
+
+	// Allocate memory for temp strings
+	char *keys = (char*) malloc(required_buffer_size_keys* sizeof(char));
+	char *values = (char*) malloc(required_buffer_size_values* sizeof(char));
+
+	int keys_offset = 0;
+	int values_offset = 0;
+	for (int i = 0; i < table->num_entries; i++)
+	{
+		memcpy(keys + keys_offset, table->entries[i].key.bytes, table->entries[i].key.len);
+		keys_offset += table->entries[i].key.len;
+		keys[keys_offset] = ';';
+		keys_offset++;
+		memcpy(values + values_offset, table->entries[i].value.value.bytes.bytes, table->entries[i].value.value.bytes.len);
+		values_offset += table->entries[i].value.value.bytes.len;
+		values[values_offset] = ';';
+		values_offset++;
+	}
+
+	// add Null-terminate the C string
+	keys[keys_offset - 1] = '\0';
+	values[values_offset - 1] = '\0';
+
+	copyStringToLStrHandle(keys, cheaders_key);
```

Comment:
what's the rational to split keys and values? any reason to favorite over `key=value;key=value` data structure?

Reply:
The simple reason is that I started with collecting keys only, and then I copied the method to support values.  I also think this way is more clear and easier to read.

Reply:
the risk though is that any issue with ordering, will create mess

Reply:
this solution is no longer valid, so I will resolve this conversation